cmd/vic-machine/validate/validator.go

```diff
@@ -45,314 +51,577 @@ type Validator struct {
-		return err
+func (v *Validator) suggestComputeResource(path string) {
+	log.Info("Suggesting valid values for --compute-resource")
+	numPools := 0
+	for ; numPools == 0; numPools = v.suggestResourcePools(path) {
+		// back up the path until we find a pool
+		newpath := filepath.Dir(path)
+		if newpath == path {
+			break
+		}
+		path = newpath
 	}
-	path, name, err = v.getNetworkPath(network)
-	if err != nil {
-		err = errors.Errorf("Failed to get bridge network path: %s", err)
-		return err
+
+	if numPools != 0 {
+		// we've made a recommendation - return
+		return
 	}
-	v.BridgeNetworkName = name
-	vchConfig.BridgeNetwork = name
-	v.BridgeNetworkPath = path
-	vchConfig.Networks["bridge"] = &metadata.NetworkInfo{
-		PortGroup:     network,
-		PortGroupName: name,
-		PortGroupRef:  network.Reference(),
-		InventoryPath: path,
+
+	// Backing all the way up didn't help
+	log.Info("Failed to find resource pool in the provided path. Showing all resource pools.")
+	numPools = v.suggestAllResourcePools()
+	if numPools == 0 {
+		log.Info("No resource pools found")
 	}
+}
 
-	// client network
-	network, err = v.Session.Finder.NetworkOrDefault(v.Context, v.ExternalNetworkPath)
-	if err != nil {
-		err = errors.Errorf("Failed to get external network: %s", err)
-		return err
+func (v *Validator) suggestAllResourcePools() int {
+	return v.suggestResourcePools("*")
+
+}
+
+func (v *Validator) suggestResourcePools(path string) int {
+	var numPools int
+
+	// list pools in path
+	numPools = v.listResourcePools(path)
+	if numPools > 0 {
+		return numPools
 	}
-	path, name, err = v.getNetworkPath(network)
-	if err != nil {
-		err = errors.Errorf("Failed to get client network path: %s", err)
-		return err
-	}
-	v.ExternalNetworkPath = path
-	vchConfig.Networks["client"] = &metadata.NetworkInfo{
-		PortGroup:     network,
-		PortGroupName: name,
-		PortGroupRef:  network.Reference(),
-		InventoryPath: path,
-	}
-
-	// management network
-	if v.ManagementNetworkName != "" {
-		network, err = v.Session.Finder.Network(v.Context, v.ManagementNetworkPath)
-		if err != nil {
-			err = errors.Errorf("Failed to get management network: %s", err)
-			return err
+
+	// no pools in path, but if path is cluster, list pools in cluster
+	clusters, _ := v.Session.Finder.ComputeResourceList(v.Context, path)
+	if clusters != nil {
+		for _, c := range clusters {
+			children := fmt.Sprintf("%s/Resources/*", c.InventoryPath)
+			log.Infof("  %s/Resources", c.InventoryPath) // Suggest `/Resources`
+			numPools += v.listResourcePools(children)
 		}
-		path, name, err = v.getNetworkPath(network)
-		if err != nil {
-			err = errors.Errorf("Failed to get management network path: %s", err)
-			return err
+	}
+
+	return numPools
+}
+
+func (v *Validator) listResourcePools(path string) int {
+	pools, _ := v.Session.Finder.ResourcePoolList(v.Context, path+"/*")
+	if len(pools) > 0 {
+		for _, p := range pools {
+			log.Infof("    %s", p.InventoryPath)
```

Comment:
Assume we'll accept relative resource pool path, instead inventory path, do we need to change the suggested resource pool format here?